### ISSUE TITLE
Hardware Connection Status

### ISSUE THREAD

**Issue by crisment** [cwhitney/GoldsprintsFx#6]
Hi
I could not run the SilverSprint Software with the Hardware. On the Settings tap under hardware connection status there is a failure. It blink and says "Version Unknow".

I did Update the firmware to basic_mrg and also use ss_basic sketch file to run the software.
I still have the same failure.

My board is a Arduino Duemilanove ATmega328

Is there anything i need to do ?

**Comment by mahroch** [comment]
Hi exactly the same with me here - same problem, same board. no solution for now :(

**Comment by crisment** [comment]
I found a solution. I bought a new Arduino Uno ATmega328 board and replaced it. After the first try it worked. Maybe there was an old chip that was not supported.
Anyway here it work now.

Best Regards
chris

**Comment by cwhitney** [comment]
If you are getting the version unknown, it means that it is finding the arduino, but it's not getting a response about the version.  The version was added in the latest release, so if you haven't manually put the latest firmware on the arduino, this is the expected behavior.  Working on getting auto-updating and better messaging working for the next release.

**Comment by jonfen** [comment]
@cwhitney opensprints shipped with either the official Arduino Duemilanove or a clone until December 2014, at which point we switched to the Arduino Uno rev3.  @crisment and @mahroch's issue might indeed have to do with the board type.  I still have an official Arduino Duemilanove and a clone so I can test those outputs.  It would be great if when an official Arduino Duemilanove or a clone is detected an upgrade to the Arduino Uno rev3 is recommended.

**Comment by mahroch** [comment]
I upgraded to Arduino Uno rev 3 and everything works like charm. The only "issue" was that new arduino uno doesn't fit to the screws in the green mainboard of opensprints hardware - I had to remove one of them. 

But connecting and all other work is completely painless, not like the older duemillenueve. I will upgrade soon all the rest of my hardware (i have 3 sets).

**Comment by crisment** [comment]
Yes i did have the same here. But I think that board will stick only with 2 screws ;-)


Von: mahroch [mailto:notifications@github.com]
Gesendet: Mittwoch, 7. Juni 2017 22:01
An: cwhitney/SilverSprint <SilverSprint@noreply.github.com>
Cc: Christian Buchner <ch.buchner@srmedia.ch>; Mention <mention@noreply.github.com>
Betreff: Re: [cwhitney/SilverSprint] Hardware Connection Status (#6)


I upgraded to Arduino Uno rev 3 and everything works like charm. The only "issue" was that new arduino uno doesn't fit to the screws in the green mainboard of opensprints hardware - I had to remove one of them.

But connecting and all other work is completely painless, not like the older duemillenueve. I will upgrade soon all the rest of my hardware (i have 3 sets).

-

**Comment by jonfen** [comment]
@mahroch and @crisment, that is correct you don't need the 3rd screw/standoff when using the Arduino Uno rev3, we now ship with 2.

**Comment by mattboulanger** [comment]
I'm playing with a cheap Uno clone. (HiLetgo UNO R3 Board ATmega328P CH340) I have been successful uploading the sketch with no errors but still get the "version unknown" flashing checkmark/x error in Silver Sprint. 

If it helps, when I ask for "board info," this is what I get: 

BN: Unknown board
VID: 1A86
PID: 7523
SN: Upload any sketch to obtain it

Any insight?

**Comment by jonfen** [comment]
We struggled with clones.  Sometimes the official drivers don't support
them, so you need other drivers

**Comment by cwhitney** [comment]
I'll try to order a clone and see if I can't get them working. I think it should be possible.

**Comment by mattboulanger** [comment]
Much appreciated. I'm not opposed to buying the real thing and trying that out, but I figured for a few bucks I'd give it a whirl on a clone. My limited googling of the issue with cheap clones reveals answers related to the CH40 chip and the need for alternative drivers, but most people are having trouble getting the computer to see the board at all until they change out drivers. I'm able to upload the sketch without errors, so I wonder if there's something about the board that it doesn't respond with a version or doesn't respond correctly. Forgive me if I'm revealing a lot of ignorance- I'm more of a bicycle guy than a programming guy!

**Comment by cwhitney** [comment]
Currently the best cross-platform way I've found to identify an arduino is to use it's port description. See here: https://github.com/cwhitney/SilverSprint/blob/master/src/data/SerialReader.cpp#L80

For official arduino's they identify with "arduino" in the name, but I'm guessing unofficial ones don't.  We should put in some more defaults to support common clones, and also allow you to manually pick an arbitrary device.
